torch/csrc/jit/passes/common_subexpression_elimination.cpp

```diff
@@ -8,6 +8,16 @@
 
 namespace torch { namespace jit {
 
+namespace {
+
+bool tensorEqual(const at::Tensor& lhs, const at::Tensor& rhs) {
+  return &lhs.type() == &rhs.type() && lhs.equal(rhs);
```

Comment:
Are we sure ATen's equality test shouldn't also handle this?

Reply:
I wasn't sure if they can be compared, so I used the addresses

Reply:
My point was that, if this wasn't already the semantics of ATen equality, then maybe we should consider adding it to the ATen API. But I'd just file an issue for it, if you agreed with me